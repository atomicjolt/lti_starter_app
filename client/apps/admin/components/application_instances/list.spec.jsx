import React from 'react';
import TestRenderer from 'react-test-renderer';
import List from './list';

describe('application instances list', () => {
  let result;
  let props;

  beforeEach(() => {
    props = {
      applicationInstances: [
        {
          id: 4,
          application_id: 7,
        },
      ],
      settings: {},
      application: {},
      canvasOauthURL: 'https://example.com',
      disableApplicationInstance: () => {},
      deleteApplicationInstance: () => {},
      saveApplicationInstance: () => {},
      sites: { 1: { id: 1, url: 'http://www.example.com' } },
      currentSortColumn: '',
      currentSortDirection: '',
      setSort: () => {},
    };
    result = TestRenderer.create(<List {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });
});
