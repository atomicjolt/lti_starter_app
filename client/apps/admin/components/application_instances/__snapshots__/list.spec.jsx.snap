// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application instances list matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    application={Object {}}
    applicationInstances={
      Array [
        Object {
          "application_id": 7,
          "id": 4,
        },
      ]
    }
    canvasOauthURL="https://example.com"
    deleteApplicationInstance={[Function]}
    disableApplicationInstance={[Function]}
    saveApplicationInstance={[Function]}
    settings={Object {}}
    sites={
      Object {
        "1": Object {
          "id": 1,
          "url": "http://www.example.com",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <thead>
          <tr>
            <th>
              <span>
                LTI KEY
              </span>
            </th>
            <th>
              <span>
                DOMAIN
              </span>
            </th>
            <th>
              <span>
                SETTINGS
              </span>
            </th>
            <th>
              <span>
                CONFIG XML
              </span>
            </th>
            <th>
              <span>
                ENABLED
              </span>
            </th>
            <th>
              <span>
                TOKEN
              </span>
            </th>
            <th>
              <span>
                AUTHENTICATIONS
              </span>
            </th>
            <th>
              <span>
                CREATED
              </span>
            </th>
            <th />
          </tr>
        </thead>,
        <tbody>
          <ListRow
            application={Object {}}
            applicationInstance={
              Object {
                "application_id": 7,
                "id": 4,
              }
            }
            canvasOauthURL="https://example.com"
            delete={[Function]}
            disable={[Function]}
            save={[Function]}
            settings={Object {}}
            sites={
              Object {
                "1": Object {
                  "id": 1,
                  "url": "http://www.example.com",
                },
              }
            }
          />
        </tbody>,
      ],
      "className": "c-table c-table--instances",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <th>
              <span>
                LTI KEY
              </span>
            </th>
            <th>
              <span>
                DOMAIN
              </span>
            </th>
            <th>
              <span>
                SETTINGS
              </span>
            </th>
            <th>
              <span>
                CONFIG XML
              </span>
            </th>
            <th>
              <span>
                ENABLED
              </span>
            </th>
            <th>
              <span>
                TOKEN
              </span>
            </th>
            <th>
              <span>
                AUTHENTICATIONS
              </span>
            </th>
            <th>
              <span>
                CREATED
              </span>
            </th>
            <th />
          </tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                <span>
                  LTI KEY
                </span>
              </th>,
              <th>
                <span>
                  DOMAIN
                </span>
              </th>,
              <th>
                <span>
                  SETTINGS
                </span>
              </th>,
              <th>
                <span>
                  CONFIG XML
                </span>
              </th>,
              <th>
                <span>
                  ENABLED
                </span>
              </th>,
              <th>
                <span>
                  TOKEN
                </span>
              </th>,
              <th>
                <span>
                  AUTHENTICATIONS
                </span>
              </th>,
              <th>
                <span>
                  CREATED
                </span>
              </th>,
              <th />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  LTI KEY
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "LTI KEY",
                },
                "ref": null,
                "rendered": "LTI KEY",
                "type": "span",
              },
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  DOMAIN
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "DOMAIN",
                },
                "ref": null,
                "rendered": "DOMAIN",
                "type": "span",
              },
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  SETTINGS
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "SETTINGS",
                },
                "ref": null,
                "rendered": "SETTINGS",
                "type": "span",
              },
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  CONFIG XML
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "CONFIG XML",
                },
                "ref": null,
                "rendered": "CONFIG XML",
                "type": "span",
              },
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  ENABLED
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ENABLED",
                },
                "ref": null,
                "rendered": "ENABLED",
                "type": "span",
              },
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  TOKEN
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "TOKEN",
                },
                "ref": null,
                "rendered": "TOKEN",
                "type": "span",
              },
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  AUTHENTICATIONS
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "AUTHENTICATIONS",
                },
                "ref": null,
                "rendered": "AUTHENTICATIONS",
                "type": "span",
              },
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  CREATED
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "CREATED",
                },
                "ref": null,
                "rendered": "CREATED",
                "type": "span",
              },
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ListRow
              application={Object {}}
              applicationInstance={
                Object {
                  "application_id": 7,
                  "id": 4,
                }
              }
              canvasOauthURL="https://example.com"
              delete={[Function]}
              disable={[Function]}
              save={[Function]}
              settings={Object {}}
              sites={
                Object {
                  "1": Object {
                    "id": 1,
                    "url": "http://www.example.com",
                  },
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "instance_0",
            "nodeType": "class",
            "props": Object {
              "application": Object {},
              "applicationInstance": Object {
                "application_id": 7,
                "id": 4,
              },
              "canvasOauthURL": "https://example.com",
              "delete": [Function],
              "disable": [Function],
              "save": [Function],
              "settings": Object {},
              "sites": Object {
                "1": Object {
                  "id": 1,
                  "url": "http://www.example.com",
                },
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "tbody",
      },
    ],
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <th>
                <span>
                  LTI KEY
                </span>
              </th>
              <th>
                <span>
                  DOMAIN
                </span>
              </th>
              <th>
                <span>
                  SETTINGS
                </span>
              </th>
              <th>
                <span>
                  CONFIG XML
                </span>
              </th>
              <th>
                <span>
                  ENABLED
                </span>
              </th>
              <th>
                <span>
                  TOKEN
                </span>
              </th>
              <th>
                <span>
                  AUTHENTICATIONS
                </span>
              </th>
              <th>
                <span>
                  CREATED
                </span>
              </th>
              <th />
            </tr>
          </thead>,
          <tbody>
            <ListRow
              application={Object {}}
              applicationInstance={
                Object {
                  "application_id": 7,
                  "id": 4,
                }
              }
              canvasOauthURL="https://example.com"
              delete={[Function]}
              disable={[Function]}
              save={[Function]}
              settings={Object {}}
              sites={
                Object {
                  "1": Object {
                    "id": 1,
                    "url": "http://www.example.com",
                  },
                }
              }
            />
          </tbody>,
        ],
        "className": "c-table c-table--instances",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                <span>
                  LTI KEY
                </span>
              </th>
              <th>
                <span>
                  DOMAIN
                </span>
              </th>
              <th>
                <span>
                  SETTINGS
                </span>
              </th>
              <th>
                <span>
                  CONFIG XML
                </span>
              </th>
              <th>
                <span>
                  ENABLED
                </span>
              </th>
              <th>
                <span>
                  TOKEN
                </span>
              </th>
              <th>
                <span>
                  AUTHENTICATIONS
                </span>
              </th>
              <th>
                <span>
                  CREATED
                </span>
              </th>
              <th />
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  <span>
                    LTI KEY
                  </span>
                </th>,
                <th>
                  <span>
                    DOMAIN
                  </span>
                </th>,
                <th>
                  <span>
                    SETTINGS
                  </span>
                </th>,
                <th>
                  <span>
                    CONFIG XML
                  </span>
                </th>,
                <th>
                  <span>
                    ENABLED
                  </span>
                </th>,
                <th>
                  <span>
                    TOKEN
                  </span>
                </th>,
                <th>
                  <span>
                    AUTHENTICATIONS
                  </span>
                </th>,
                <th>
                  <span>
                    CREATED
                  </span>
                </th>,
                <th />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    LTI KEY
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "LTI KEY",
                  },
                  "ref": null,
                  "rendered": "LTI KEY",
                  "type": "span",
                },
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    DOMAIN
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "DOMAIN",
                  },
                  "ref": null,
                  "rendered": "DOMAIN",
                  "type": "span",
                },
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    SETTINGS
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "SETTINGS",
                  },
                  "ref": null,
                  "rendered": "SETTINGS",
                  "type": "span",
                },
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    CONFIG XML
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "CONFIG XML",
                  },
                  "ref": null,
                  "rendered": "CONFIG XML",
                  "type": "span",
                },
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    ENABLED
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ENABLED",
                  },
                  "ref": null,
                  "rendered": "ENABLED",
                  "type": "span",
                },
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    TOKEN
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "TOKEN",
                  },
                  "ref": null,
                  "rendered": "TOKEN",
                  "type": "span",
                },
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    AUTHENTICATIONS
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "AUTHENTICATIONS",
                  },
                  "ref": null,
                  "rendered": "AUTHENTICATIONS",
                  "type": "span",
                },
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    CREATED
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "CREATED",
                  },
                  "ref": null,
                  "rendered": "CREATED",
                  "type": "span",
                },
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ListRow
                application={Object {}}
                applicationInstance={
                  Object {
                    "application_id": 7,
                    "id": 4,
                  }
                }
                canvasOauthURL="https://example.com"
                delete={[Function]}
                disable={[Function]}
                save={[Function]}
                settings={Object {}}
                sites={
                  Object {
                    "1": Object {
                      "id": 1,
                      "url": "http://www.example.com",
                    },
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "instance_0",
              "nodeType": "class",
              "props": Object {
                "application": Object {},
                "applicationInstance": Object {
                  "application_id": 7,
                  "id": 4,
                },
                "canvasOauthURL": "https://example.com",
                "delete": [Function],
                "disable": [Function],
                "save": [Function],
                "settings": Object {},
                "sites": Object {
                  "1": Object {
                    "id": 1,
                    "url": "http://www.example.com",
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
